gtk4/src/subclass/im_context.rs

```diff
@@ -0,0 +1,512 @@
+use crate::{IMContext, Widget};
+use glib::subclass::object::ObjectImpl;
+use glib::subclass::prelude::*;
+use glib::translate::*;
+use glib::{Cast, GString, IsA, Object};
+use pango::AttrList;
+
+pub trait IMContextImpl: IMContextImplExt + ObjectImpl {
+    fn commit(&self, im_context: &Self::Type, string: &str) {
+        self.parent_commit(im_context, string)
+    }
+    fn delete_surrounding(&self, im_context: &Self::Type, offset: i32, n_chars: i32) -> bool {
+        self.parent_delete_surrounding(im_context, offset, n_chars)
+    }
+    fn filter_keypress(&self, im_context: &Self::Type, event: &gdk::Event) -> bool {
+        self.parent_filter_keypress(im_context, event)
+    }
+    fn focus_in(&self, im_context: &Self::Type) {
+        self.parent_focus_in(im_context)
+    }
+    fn focus_out(&self, im_context: &Self::Type) {
+        self.parent_focus_out(im_context)
+    }
+    fn get_preedit_string(&self, im_context: &Self::Type) -> (GString, AttrList, i32) {
+        self.parent_get_preedit_string(im_context)
+    }
+    fn get_surrounding(&self, im_context: &Self::Type) -> Option<(GString, i32)> {
+        self.parent_get_surrounding(im_context)
+    }
+    fn preedit_changed(&self, im_context: &Self::Type) {
+        self.parent_preedit_changed(im_context)
+    }
+    fn preedit_end(&self, im_context: &Self::Type) {
+        self.parent_preedit_end(im_context)
+    }
+    fn preedit_start(&self, im_context: &Self::Type) {
+        self.parent_preedit_start(im_context)
+    }
+    fn reset(&self, im_context: &Self::Type) {
+        self.parent_reset(im_context)
+    }
+    fn retrieve_surrounding(&self, im_context: &Self::Type) -> bool {
+        self.parent_retrieve_surrounding(im_context)
+    }
+    fn set_client_widget<P: IsA<Widget>>(&self, im_context: &Self::Type, widget: Option<&P>) {
+        self.parent_set_client_widget(im_context, widget)
+    }
+    fn set_cursor_location(&self, im_context: &Self::Type, area: &mut gdk::Rectangle) {
```

Comment:
```suggestion
    fn set_cursor_location(&self, im_context: &Self::Type, area: &gdk::Rectangle) {
```

etc according to the docs